src/Microsoft.ML.Data/Transforms/CopyColumnsTransform.cs

```diff
@@ -163,4 +169,189 @@ protected override Delegate GetGetterCore(IChannel ch, IRow input, int iinfo, ou
             return (Delegate)methodInfo.Invoke(input, new object[] { col });
         }
     }
+
+    public sealed class CopyColumnsEstimator : IEstimator<CopyColumnsTransformer>
+    {
+        private readonly (string Source, string Name)[] _columns;
+        private readonly IHost _host;
+
+        public CopyColumnsEstimator(IHostEnvironment env, string input, string output)
+        {
+            Contracts.CheckNonWhiteSpace(input, nameof(input));
+            Contracts.CheckNonWhiteSpace(output, nameof(output));
+            _columns = new (string, string)[1] { (input, output) };
+            _host = env.Register("CopyColumnsEstimator");
+        }
+
+        public CopyColumnsEstimator(IHostEnvironment env, (string Source, string Name)[] columns)
```

Comment:
>CopyColumnsEstimator [](start = 15, length = 20)

params?

Reply:
what params?

---
In reply to: [211790834](https://github.com/dotnet/machinelearning/pull/706#discussion_r211790834) [](ancestors = 211790834)

Reply:
make these a params (Source, Name)[] parameter

---
In reply to: [212126225](https://github.com/dotnet/machinelearning/pull/706#discussion_r212126225) [](ancestors = 212126225,211790834)